tests/prow_ci.sh

```diff
@@ -30,3 +30,147 @@ override_rosacli_build () {
+  # configure aws credetials
+  AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+  if [[ -f "${AWSCRED}" ]]; then
+    export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+    export AWS_DEFAULT_REGION="${REGION}"
+  else
+    echo "Did not find compatible cloud provider cluster_profile"
+    exit 1
+  fi
+
+  # configure shared vpc credentials
+  if [[ -f ${CLUSTER_PROFILE_DIR}/.awscred_shared_account ]];then
+    echo "[CI] Got awscred_shared_account and set it to env variable SHARED_VPC_AWS_SHARED_CREDENTIALS_FILE"
+    export SHARED_VPC_AWS_SHARED_CREDENTIALS_FILE=${CLUSTER_PROFILE_DIR}/.awscred_shared_account
+  fi
+}
+
+# configure rosa login
+# CLUSTER_PROFILE_DIR is global env, no need to set
+# OCM_LOGIN_ENV need to be specified on job level
+rosa_login () {
+  echo "Running rosa login based on env ${OCM_LOGIN_ENV}"
+  OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+  if [[ ! -z "${OCM_TOKEN}" ]]; then
+    echo "[CI] Login into ${OCM_LOGIN_ENV}"
+    rosa login --env "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+    ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+  else
+    echo "Cannot login! You need to specify the offline token OCM_TOKEN!"
+    exit 1
+  fi
+}
+
+# generate_label_filter_switch is used to generate label-filter for the test run
+# IMPORTANCE is step env for job running
+# LABEL_FILTER is global env setting but it can be override by parameter
+# if there is parameter when call this function, it will override
+# Need to clarify the LABEL_FILTER_SWITCH in steps script 
+generate_label_filter_switch () {
+  label_filter=${LABEL_FILTER}
+  if [ ! -z "$1" ]; then
+    echo "[CI] Got parameter label filter: $1 . Override the global setting now."
+    label_filter=$1
+  fi
+  if [[ ! -z "$IMPORTANCE" ]]; then
+    label_filter="$label_filter&&${IMPORTANCE}" 
+  fi
+  LABEL_FILTER_SWITCH="--ginkgo.label-filter '${label_filter}'"
+}
+
+# generate_junit is used to generate junit file path
+# RUN_TIME needs to be set in the step or job level
+# SHARED_DIR is global env for the CI run, no need to set
+# TEST_PROFILE is optional in job level
+# JUNIT_XML need to be clarified in the step definition
+# JUNIT_TEMP_DIR need to be clarified in the step definition
+# there is usage accepts a parameter to override the default usage
+generate_junit () {
+  usage="e2e"
+  if [ ! -z $1 ];then
+    usage="$1"
+  fi
+  JUNIT_TEMP_DIR=$(mktemp -d)
+  junit_file_name="rosa-$usage"
+  if [ ! -z "${TEST_PROFILE}" ]; then
+    junit_file_name="${junit_file_name}-${TEST_PROFILE}"
+  fi
+  if [ ! -z "${RUN_TIME}" ]; then
+    junit_file_name="${junit_file_name}-${RUN_TIME}"
+  fi
+
+  JUNIT_XML="${JUNIT_TEMP_DIR}/${junit_file_name}.xml"
+
+  echo "[CI] the junit temp dir is $JUNIT_TEMP_DIR and junit file will be $JUNIT_XML" 
+}
+
+# extract_existing_junit is used to extract existing junit file from previous steps
+# JUNIT_TEMP_DIR should be clarified in step and generated by generate_junit
+extract_existing_junit () {
```

Comment:
In the future, I plan to combine the runtime steps into one and use a variable instead of multiple steps to improve the flexibility of the test chain. If that is true, we do not need to extract/package the `junit.tar.gz` in each step. 

Besides, for the current status, I think it is better to generate a different same-prefix `*.tar.gz` and let the reported step do one-time extract/package.

Reply:
Let's implement as step by step. This is the final expected implementation way. But it's long term resolution.